contracts/ethereum/src/PaymentRegistry.sol

```diff
@@ -30,84 +39,125 @@ contract PaymentRegistry is Initializable, OwnableUpgradeable, UUPSUpgradeable {
+        require(msg.value > 0, "Invalid amount, should be higher than 0.");
+
+        bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, msg.value, chainId)); //200 gas
+        require(transfers[index].isUsed == false, "Transfer already processed."); //3000 gas
 
-        bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount));
-        require(transfers[index].isUsed == false, "Transfer already processed.");
+        // transfers[index] = TransferInfo({destAddress: destAddress, amount: msg.value, isUsed: true, chainId: chainId}); //65000 gas
+        // 64900 gas:
+        transfers[index].destAddress = destAddress;
+        transfers[index].amount = msg.value;
+        transfers[index].isUsed = true;
+        transfers[index].chainId = chainId;
 
-        transfers[index] = TransferInfo({destAddress: destAddress, amount: amount, isUsed: true});
 
-        (bool success,) = payable(address(uint160(destAddress))).call{value: msg.value}("");
+        (bool success,) = payable(address(uint160(destAddress))).call{value: msg.value}(""); //34000 gas
+        // (bool success,) = payable(marketMaker).call{value: msg.value}(""); //32000 gas //to implement this, address must be changed to from uint256 to addr
 
         require(success, "Transfer failed.");
-        emit Transfer(orderId, msg.sender, transfers[index]);
+        emit Transfer(orderId, msg.sender, transfers[index]); //3000 gas
     }
 
+//TODO change name to claimPaymentStarknet
     function claimPayment(uint256 orderId, uint256 destAddress, uint256 amount) external payable onlyOwnerOrMM {
-        bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount));
+        bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount, Chain.Starknet));
         TransferInfo storage transferInfo = transfers[index];
         require(transferInfo.isUsed == true, "Transfer not found.");
 
-        uint256[] memory payload = new uint256[](5);
+        uint256[] memory payload = new uint256[](5); //TODO why array of 256 if then filled with 128?
         payload[0] = uint128(orderId); // low
         payload[1] = uint128(orderId >> 128); // high
         payload[2] = transferInfo.destAddress;
         payload[3] = uint128(amount); // low
         payload[4] = uint128(amount >> 128); // high
-        
+            
         _snMessaging.sendMessageToL2{value: msg.value}(
-            _snEscrowAddress,
-            _snEscrowClaimPaymentSelector,
+            StarknetEscrowAddress,
+            StarknetEscrowClaimPaymentSelector,
             payload);
+
+        emit ClaimPayment(transferInfo);
     }
 
-    function setEscrowAddress(uint256 snEscrowAddress) external onlyOwner {
-        _snEscrowAddress = snEscrowAddress;
-        emit ModifiedEscrowAddress(snEscrowAddress);        
+// why is destAddress a uint256? should be address?
+// oh, because sn address is not "address" typew]
```

Comment:
The comment was made in zksync because the problem was in that method. Anyways, the comment was removed because it was a mistake, destAddress should be address type (and it is in PR #183 ) because destAddress is the destination address of the user's funds in L1, which is always type address.